node/bindnode/repr.go

```diff
@@ -557,22 +557,50 @@ func (w *_assemblerRepr) AssignString(s string) error {
 			if member.Name() != name {
 				continue
 			}
-			valPtr := reflect.New(goTypeString)
-			valPtr.Elem().SetString(s)
-			unionSetMember(w.val, idx, valPtr)
-			return nil
+
+			// TODO: DRY with behavior below.
+			w2 := *w
+			goType := w.val.Field(idx).Type().Elem()
+			valPtr := reflect.New(goType)
+			w2.val = valPtr.Elem()
+			w2.schemaType = member
+			w2.finish = func() error {
+				if w.finish != nil {
+					if err := w.finish(); err != nil {
+						return err
+					}
+				}
+				unionSetMember(w.val, idx, valPtr)
+				return nil
+			}
+
+			return w2.AssignString(s)
```

Comment:
have you seen asKinded? because this looks extremely similar. we use asKinded in AsString on kinded unions, so it makes sense to use it in AssignString too.

The other AsT/AssignT lack the support for kinded unions simply because I haven't gotten to propagating the code.

Reply:
Oh, nice.  I had not seen it; that works perfectly.

Reply:
It's possible some of the code could do with more reuse. I haven't paid *too* much attention to that just yet. I find a bit of copy-pasting to be necessary for the sake of quick iteration - otherwise I spend extra time extracting and inlining common functions :)